sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorReceiver.java

```diff
@@ -95,59 +91,76 @@ protected ReactorReceiver(AmqpConnection amqpConnection, String entityPath, Rece
+                                    sink.error(new IllegalStateException(
+                                        "Cannot decode delivery when ReactorReceiver instance is closed."));
+                                    return;
+                                }
+
+                                final Message message = decodeDelivery(delivery);
+                                if (metricsProvider.isPrefetchedSequenceNumberEnabled()) {
+                                    Long seqNo = getSequenceNumber(message);
+                                    if (seqNo != null) {
+                                        lastSequenceNumber.set(seqNo);
+                                    }
+                                }
 
-        // Delivered messages are not published on another scheduler because we want the settlement method that happens
-        // in decodeDelivery to take place and since proton-j is not thread safe, it could end up with hundreds of
-        // backed up deliveries waiting to be settled. (Which, consequently, ends up in a FAIL_OVERFLOW error from
-        // the handler.
-        this.messagesProcessor = this.handler.getDeliveredMessages()
-            .flatMap(delivery -> {
-                return Mono.create(sink -> {
-                    try {
-                        this.dispatcher.invoke(() -> {
-                            if (isDisposed()) {
-                                sink.error(new IllegalStateException(
-                                    "Cannot decode delivery when ReactorReceiver instance is closed."));
-                                return;
-                            }
-
-                            final Message message = decodeDelivery(delivery);
-                            if (metricsProvider.isPrefetchedSequenceNumberEnabled()) {
-                                Long seqNo = getSequenceNumber(message);
-                                if (seqNo != null) {
-                                    lastSequenceNumber.set(seqNo);
+                                final int creditsLeft = receiver.getRemoteCredit();
+
+                                if (creditsLeft > 0) {
+                                    sink.success(message);
+                                    return;
                                 }
-                            }
 
-                            final int creditsLeft = receiver.getRemoteCredit();
+                                final Supplier<Integer> supplier = creditSupplier.get();
+                                final Integer credits = supplier.get();
+
+                                if (credits != null && credits > 0) {
+                                    logger.atVerbose()
+                                        .addKeyValue("credits", credits)
+                                        .log("Adding credits.");
+                                    receiver.flow(credits);
+                                } else {
+                                    logger.atVerbose()
```

Comment:
I don't think this log is necessary.

Reply:
Yea, it’s v1 code and not triggered when running under v2, but I think we can remove the log though.